apps/astarte_appengine_api/test/astarte_appengine_api/v2_group_test.exs

```diff
@@ -0,0 +1,129 @@
+        assert [error] === json_response(response, 422)["errors"]["group_name"]
+      end
+    end
+
+    property "devices list empty", %{auth_conn: auth_conn, keyspace: keyspace} do
+      check all group_name <- GroupGenerator.name() do
+        params = %{
+          "group_name" => group_name,
+          "devices" => []
+        }
+
+        response = post(auth_conn, groups_path(auth_conn, :create, keyspace), data: params)
+
+        assert ["should have at least 1 item(s)"] ===
+                 json_response(response, 422)["errors"]["devices"]
+      end
+    end
+
+    property "device does not exist", %{auth_conn: auth_conn, keyspace: keyspace} do
+      check all group_name <- GroupGenerator.name(),
+                devices <- list_of(DeviceGenerator.encoded_id(), min_length: 1) do
+        params = %{
+          "group_name" => group_name,
+          "devices" => devices
+        }
+
+        response = post(auth_conn, groups_path(auth_conn, :create, keyspace), data: params)
+
+        assert ["must exist (#{Enum.at(devices, 0)} not found)"] ===
+                 json_response(response, 422)["errors"]["devices"]
+      end
+    end
+
+    property "rejects an already existing group", %{
+      auth_conn: auth_conn,
+      cluster: cluster,
+      keyspace: keyspace,
+      interfaces: interfaces,
+      devices: devices,
+      groups: groups
+    } do
+      device_ids = Enum.map(devices, & &1.encoded_id)
+      existing_group_name = Enum.at(groups, 0).name
+
+      params = %{
+        "group_name" => existing_group_name,
+        "devices" => device_ids
+      }
+
+      response = post(auth_conn, groups_path(auth_conn, :create, keyspace), data: params)
+      assert "Group already exists" === json_response(response, 409)["errors"]["detail"]
+    end
+
+    property "creates groups with valid parameters", %{
+      auth_conn: auth_conn,
+      cluster: cluster,
+      keyspace: keyspace,
+      interfaces: interfaces,
+      devices: devices,
+      groups: groups
+    } do
+      device_ids = Enum.map(devices, & &1.encoded_id)
+      old_group_names = Enum.map(groups, & &1.name)
+
+      check all group_name <-
+                  filter(GroupGenerator.name(), fn name -> name not in old_group_names end) do
+        params = %{
+          "group_name" => group_name,
+          "devices" => device_ids
+        }
+
+        response = post(auth_conn, groups_path(auth_conn, :create, keyspace), data: params)
+        assert params === json_response(response, 201)["data"]
+        response = get(auth_conn, groups_path(auth_conn, :show, keyspace, group_name))
+        assert group_name === json_response(response, 200)["data"]["group_name"]
+
+        for device <- device_ids do
+          {:ok, %DeviceStatus{groups: groups}} = Device.get_device_status!(keyspace, device)
+          assert group_name in groups
+          # TODO right way
+          # assert [group_name] === groups -- old_group_names
```

Comment:
If we're just testing that a group is being created, why checking that every device in the group has it in its status? This is probably a (slightly) different property that should be checked in another test